### DISCUSSION TITLE
Setup Guide

### DISCUSSION THREAD

**Discussion by cholzer79** [joseantmazonsb/linguard#102] [Q&A]
I think that a detailed setup guide is needed. 😅

I eventually got it working, let me share how - maybe others will find it helpful. (disclaimer, it is entirely possible that I did some things horribly wrong!)

I am using Ubuntu 20.04.03

**prerequisites:**
the PC/VM you install this on should have a static IP

**install wireguard:**
```
sudo -s 
apt-get update && apt-get upgrade
apt-get install wireguard
```

**install linguard:**
```
mkdir .\linguard
wget https://github.com/joseantmazonsb/linguard/releases/download/v1.1.0/linguard-1.1.0.tar.gz
tar -zxvf linguard-1.1.0.tar.gz
rm linguard-1.1.0.tar.gz
chmod +x install.sh
sudo ./install.sh

systemctl start linguard.service
```
**configure linguard**
1. access the webgui https://IP:8080
2. create a user
3. set the correct endpoint (if your side has a static WAN IP then enter that, otherwise enter your FQDN or dynDNS name of the site)
4. go to system->wireguard
5. add an interface name it i.e. "wg" for wireguard and write down the port
5. start the interface 
6. next go to the port-forwarding settings in your router and then forward that port set in the interface to the IP of the server/vm that linguard/wireguard is running on
7. add a peer
8. give it a clear name like "marks_iphone", select the "wg" interface you created, enable NAT if required
9. use the QR code to add the config on your mobile phone
10. test the connection and monitor the traffic on the dashboard

**Comment by joseantmazonsb** [comment]
Sorry I'm a little bit late to the party. I just wanted to clarify that there's actually no need to run `apt update` or install `wireguard` manually, since running `install.sh` will automatically do it.
